qubesbuilder/plugins/source_deb/scripts/clamp-changelog-entry-date

```diff
@@ -30,9 +30,13 @@ if [ ! -r "$CHANGELOG_PATH" ]; then
     exit 1
 fi
 
-# get previous date
+# Get previous date from the changelog
 PREVIOUS_DATE=$(grep '^ --' "$CHANGELOG_PATH" | head -n 2 | tail -n 1 | grep -o '  .*')
 
-# replace topmost date
-sed -e "0,/^ --/s/^\( --.*\)\(  .*\)/\1$PREVIOUS_DATE/" -i "$CHANGELOG_PATH"
-touch --date="$PREVIOUS_DATE" "$CHANGELOG_PATH"
+# Use SOURCE_DATE_EPOCH if available; otherwise, use PREVIOUS_DATE
+CURRENT_DATE=$(date -u -d "@${SOURCE_DATE_EPOCH:-$PREVIOUS_DATE}" +"%Y-%m-%d %H:%M:%S")
+
+# Replace the topmost date with CURRENT_DATE
+sed -e "0,/^ --/s/^\( --.*\)\(  .*\)/\1$CURRENT_DATE/" -i "$CHANGELOG_PATH"
+touch --date="$CURRENT_DATE" "$CHANGELOG_PATH"
+
```

Comment:
I've reverted the script to its original logic, where the topmost date is replaced with the previous date, without involving $SOURCE_DATE_EPOCH, since it wasn't necessary for this case.